### ISSUE TITLE
skip twisted.words' msn tests when twisted.web is not available

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11184]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#1735|
|Type|task|
|Created|2006-05-22 00:01:20Z|

<details><summary>Searchable metadata</summary>

```
trac-id__1735 1735
type__task task
reporter__radix radix
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1148256080000000 1148256080000000
changetime__1149995104000000 1149995104000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

Tests made conditional in [branches/words-msn-http-1735](branches/words-msn-http-1735) - please review.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @jml|
|-|-|

I'll review it.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @radix|
|-|-|

Definitely the right approach to take with skipping tests en masse (at least, for the moment). 

However, some classes in test_msn (`DummySwitchboardClient` et al) inherit from `msn.*`.  This won't work if `msn` is set to `None`. e.g. 

```
===============================================================================
[ERROR]: twisted/words/test/test_msn.py

Traceback (most recent call last):
  File "/Users/jml/Projects/Twisted/trunk/twisted/trial/runner.py", line 343, in loadByName
    thing = self.findByName(name)
  File "/Users/jml/Projects/Twisted/trunk/twisted/trial/runner.py", line 251, in findByName
    return filenameToModule(name)
  File "/Users/jml/Projects/Twisted/trunk/twisted/trial/runner.py", line 54, in filenameToModule
    return _importFromFile(fn)
  File "/Users/jml/Projects/Twisted/trunk/twisted/trial/runner.py", line 73, in _importFromFile
    module = imp.load_source(moduleName, fn, fd)
  File "/Users/jml/Projects/Twisted/trunk/twisted/words/test/test_msn.py", line 99, in ?
    class DummySwitchboardClient(msn.SwitchboardClient):
exceptions.AttributeError: 'NoneType' object has no attribute 'SwitchboardClient'
-------------------------------------------------------------------------------
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @jml|
|-|-|

I've put the two classes which subclassed something from the msn module inside an if so that they are not defined if everything is going to be skipped.

Does this seem reasonable?

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @exarkun|
|-|-|

Yes. Merge away.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [17163]) Merge words-msn-http-1735

Author: exarkun
Reviewer: jml
Fixes [#11184](https://github.com/twisted/twisted/issues/11184)

Skip MSN tests which require an HTTP client if no HTTP client is available.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
